### PR TITLE
List spacing

### LINKED ISSUE #2788: [Innspill til komponent]: Endre whitespace for <List>
### Tilbakemelding

Det blir litt mye whitespace for `<List>` slik den er i dag, lenke til diskusjon: https://nav-it.slack.com/archives/C7NE7A8UF/p1710338460659119

Saken er at det er vel mye luft mellom punktene nå. Vi må finne ut hvor mye vi kan justere luften mellom listeelementene slik at det fortsatt ser ut som en liste.
Krav: det kan ikke være mindre luft enn det som er mellom linjer i teksten.

### PR DESCRIPTION
**Pull request by KenAJoh**
### Description

Resolves #2788

### Change summary

- Changed spacing between List.Item to spacing-3 from spacing 4, (3 -> 2 for small)
- Changed BodyShort -> BodyLong for List.Item

### CODE REVIEW

File: @navikt/core/css/list.css [hunk 1]
```diff
@@ -15,15 +15,15 @@
...
 
 .navds-list--small li {
-  margin-block-end: var(--a-spacing-3);
+  margin-block-end: var(--a-spacing-2);
 }
 
 .navds-list--nested li {
-  margin-block-end: var(--a-spacing-3);
+  margin-block-end: var(--a-spacing-2);
 }
```
Comment (HalvorHaugan) [inline @navikt/core/css/list.css]: Are these redundant?

```css
.navds-list--small li {
  margin-block-end: var(--a-spacing-2);
}

.navds-list--nested li {
  margin-block-end: var(--a-spacing-2);
}
```

File: @navikt/core/css/list.css
Comment (HalvorHaugan) [inline @navikt/core/css/list.css:1]: The bullet might need vertical adjustment

File: @navikt/core/css/list.css (hunk 1 above)
Comment (KenAJoh) [reply @navikt/core/css/list.css]: Fjernet dem 👍

File: @navikt/core/css/list.css
Comment (KenAJoh) [reply @navikt/core/css/list.css:1]: Fikset nå 🤞

**Review by HalvorHaugan — DISMISSED** [review]

File: .changeset/strong-bulldogs-jam.md [hunk 2]
```diff
@@ -0,0 +1,6 @@
+---
+"@navikt/ds-react": minor
+"@navikt/ds-css": minor
```
Comment (HalvorHaugan) [inline .changeset/strong-bulldogs-jam.md]: Isn't this patch?

Comment (KenAJoh) [inline .changeset/strong-bulldogs-jam.md]:
```suggestion
"@navikt/ds-react": patch
"@navikt/ds-css": patch
```

**Review by HalvorHaugan — APPROVED** [review]

**Pull request merged**